org.knime.scijava.commands/src/org/knime/scijava/commands/settings/DefaultSettingsModelTypeService.java

```diff
@@ -20,60 +25,85 @@
 @SuppressWarnings("rawtypes")
 @Plugin(type = SettingsModelTypeService.class)
 public class DefaultSettingsModelTypeService
-		extends AbstractSingletonService<SettingsModelTypePlugin>
-		implements SettingsModelTypeService {
-
-	private final Map<Class<? extends SettingsModel>, SettingsModelTypePlugin> m_pluginsByModel = new HashMap<>();
-	private final Map<Class<?>, SettingsModelTypePlugin> m_pluginsByValue = new HashMap<>();
-
-	@Override
-	public Class<SettingsModelTypePlugin> getPluginType() {
-		return SettingsModelTypePlugin.class;
-	}
-
-	@Override
-	public SettingsModelType getSettingsModelTypeFor(
-			final SettingsModel settingsModel) {
-
-		final SettingsModelTypePlugin plugin = m_pluginsByModel
-				.get(settingsModel.getClass());
-		if (plugin != null) {
-			return plugin;
-		}
-
-		for (final SettingsModelTypePlugin p : getInstances()) {
-			if (p.getSettingsModelClass().isInstance(settingsModel)) {
-				m_pluginsByModel.put(settingsModel.getClass(), p);
-				return p;
-			}
-		}
-		// nothing found
-		return null;
-	}
-
-	@Override
-	public SettingsModelType getSettingsModelTypeFor(final Class<?> value) {
-
-		// check cache
-		final SettingsModelTypePlugin plugin = m_pluginsByValue.get(value);
-		if (plugin != null) {
-			return plugin;
-		}
-
-		// check primitive conversion cache
-		final Class<?> checkValue = PrimitiveTypeUtils
-				.convertIfPrimitive(value);
-
-		// search
-		for (final SettingsModelTypePlugin<?, ?> p : getInstances()) {
-			if (p.getValueClass().isAssignableFrom(checkValue)) {
-				m_pluginsByValue.put(value, p);
-				return p;
-			}
-		}
-
-		// nothing found
-		return null;
-	}
+        extends AbstractSingletonService<SettingsModelType>
+        implements SettingsModelTypeService {
+
+    @Parameter
+    private ConverterCacheService cs;
+
+    private final Map<Class<? extends SettingsModel>, SettingsModelType> m_pluginsByModel = new HashMap<>();
+    private final Map<Class<?>, SettingsModelType> m_pluginsByValue = new HashMap<>();
+
+    @Override
+    public SettingsModelType getSettingsModelTypeFor(
+            final SettingsModel settingsModel) {
+
+        final SettingsModelType plugin = m_pluginsByModel
+                .get(settingsModel.getClass());
+        if (plugin != null) {
+            return plugin;
+        }
+
+        for (final SettingsModelType p : getInstances()) {
+            if (p.getSettingsModelClass().isInstance(settingsModel)) {
+                m_pluginsByModel.put(settingsModel.getClass(), p);
+                return p;
+            }
+        }
+        // nothing found
+        return null;
+    }
+
+    @Override
+    public SettingsModelType getSettingsModelTypeFor(final ModuleItem<?> item) {
+
+        // FIXME why do I need this check in case of scripts?... Why should you
+        // not?
```

Comment:
> This is a slight inconsistency in Scijava. What is your oppinion about this? There seems to be no documentation about the `getWidgetStyle()` method possibly returning null, so I guess the widget style should probably be initialized with `""` for DefaultMutableModuleItem?

I agree this is inconsistent. However, it goes far beyond just widget style. _Every_ attribute of the `@Parameter` annotation has non-null defaults, because Java annotations do not allow null values whatsoever. However, that is inconsistent with SJC behavior in general, where `null`—not `""`—is used to indicate no/default value/behavior.

I suppose my preferred solution to this conundrum is to change the `PluginInfo` class to translate empty strings into nulls, as perverse as that seems...